.github/workflows/testing.yaml

```diff
@@ -21,87 +21,84 @@ jobs:
-        shell: bash
-
-      - name: Cache tarantool build
-        if: startsWith(matrix.tarantool, 'debug')
-        id: cache-tnt-debug
+      - name: Cache rocks
         uses: actions/cache@v3
+        id: cache-rocks
         with:
-          path: ${{ env.TNT_DEBUG_PATH }}
-          key: cache-tnt-${{ matrix.tarantool }}${{ env.VERSION_POSTFIX }}
+          path: .rocks/
+          key: "cache-rocks-${{ matrix.tarantool }}${{ env.VERSION_POSTFIX }}"
 
-      - name: Clone tarantool ${{ matrix.tarantool }}
-        if: startsWith(matrix.tarantool, 'debug') && steps.cache-tnt-debug.outputs.cache-hit != 'true'
-        uses: actions/checkout@v3
-        with:
-          repository: tarantool/tarantool
-          ref: ${{ env.TNT_BRANCH }}
-          path: tarantool
-          fetch-depth: 0
-          submodules: true
-
-      - name: Build tarantool ${{ matrix.tarantool }}
-        if: startsWith(matrix.tarantool, 'debug') && steps.cache-tnt-debug.outputs.cache-hit != 'true'
-        run: |
-          sudo apt-get -y install git build-essential cmake make zlib1g-dev \
-            libreadline-dev libncurses5-dev libssl-dev \
-            libunwind-dev libicu-dev python3 python3-yaml \
-            python3-six python3-gevent
-          cd ${GITHUB_WORKSPACE}/tarantool
-          mkdir build && cd build
-          cmake .. -DCMAKE_BUILD_TYPE=Debug -DENABLE_DIST=ON
-          make
-          make DESTDIR=${TNT_DEBUG_PATH} install
-
-      - name: Install tarantool ${{ matrix.tarantool }}
-        if: startsWith(matrix.tarantool, 'debug')
-        run: |
-          sudo cp -rvP ${TNT_DEBUG_PATH}/usr/local/* /usr/local/
+      - name: Install requirements
+        run: make deps depname=test
+        if: steps.cache-rocks.outputs.cache-hit != 'true'
 
+      - run: echo $PWD/.rocks/bin >> $GITHUB_PATH
+
+      - run: make test
+
+  coverage:
+    # We want to run on external PRs, but not on our own internal
+    # PRs as they'll be run by the push to the branch.
+    #
+    # The main trick is described here:
+    # https://github.com/Dart-Code/Dart-Code/pull/2375
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+
+    strategy:
+      fail-fast: false
+
+    env:
+      TNT_DEBUG_PATH: /home/runner/tnt-debug
+
+    runs-on: ubuntu-22.04
+    steps:
       - name: Clone the module
         uses: actions/checkout@v3
 
+      - name: Setup tt
+        run: |
+          curl -L https://tarantool.io/release/3/installer.sh | bash
+          sudo apt install -y tt
+          tt version
+
+      - name: Install Tarantool
+        uses: ./.github/actions/install-tarantool
+        with: 
+          tarantool: '3.1'
+
       - name: Cache rocks
         uses: actions/cache@v3
         id: cache-rocks
         with:
           path: .rocks/
-          key: "cache-rocks-${{ matrix.tarantool }}${{ env.VERSION_POSTFIX }}"
+          key: "cache-rocks-3.1${{ env.VERSION_POSTFIX }}"
```

Comment:
I don't get it. None of the previous steps had installed any rocks. (And even if they are, are we really want to fixate rocks based on Tarantool binary version?)

Reply:
Delete tarantool version from the `cache` key and reworked `cache` step - moved it after rocks installation.

Reply:
Do we really need cache here? It should not take too much time to re-build rocks here.

Reply:
Agree, deleted.